public/javascript/pump/socket.js

```diff
@@ -164,14 +164,10 @@
                 _socket.retryTimer = null;
             }
 
-            if (error.code === 2000 && _socket.retryAttempts > 10) {
-                // No more attempts if all transports failed
-                if (!_socket.error) {
-                    // Show alert only one time
-                    _socket.error = error;
-                    Pump.error("Looks like your browser not support any socket transport");
-                    Pump.debug(error);
-                }
+            // No more attempts if all transports failed
+            if (error.code === 2000) {
+                Pump.error("Sorry! Your browser doesn't support realtime.");
+                Pump.debug(error);
```

Comment:
Wait, won't this code get triggered if the user goes offline? In that case the error doesn't make sense, right?

Reply:
Sorry I didn't catch this earlier :thumbsup:

Reply:
But only for the socket, but the interface still alive, and if this error happens to some user is a good idea notification and a log for try to detect if a problem of the browser or the socket.js library or other error, so the notification is necessary and will be shown in the case of that error because happen after to try to connect to the socket and after previous network connection and after the UI was loaded

Reply:
I'm having a little bit of a hard time following what you just said. I'm not suggesting we get rid of the error message, just that we fix it so it doesn't say your browser doesn't support realtime if the user is just offline. Can't we just keep a boolean indicating if there's ever been a successful connection, and if that variable is `true`, assume the user is just offline? And show them something to that effect, e.g. `You seem to be offline! Realtime is disabled.`?

Reply:
So here's the scenario I'm thinking of:

1. User visits the page online, SockJS establishes a connection with some transport (probably WebSockets on a modern browser)
2. User moves offline, potentially because they're on a mobile device, etc.
3. SockJS connection is broken and triggers this retry logic
4. All transports fail because the user is offline
5. User sees `Sorry! Your browser doesn't support realtime.` which isn't accurate

If the user visits the page and all transports immediately fail then of course we should still show this error message, since it's accurate. But in the case I've just outlined it's not.

Reply:
in socket.js the 2000 error ('All transports failed') only happens when trying the browser support. not when the user moves offline, in that case is `1000`

https://github.com/sockjs/sockjs-client/blob/436e997549f3937cc16bd5085addb218d43cfb60/lib/main.js#L328